notebooks/06-cifar10-baseline.ipynb

```diff
@@ -0,0 +1,392 @@
+{
```

Comment:
<p>can you try <code>update_bn(...)</code> in <code>on_train_end</code> hook instead of calling it explicitly after <code>trainer.fit</code>?</p><p><code>update_bn(self.datamodule.train_dataloader(), self.swa_model, device=self.device)</code></p><p> </p><p>also what's the significance of using SWA before testing?</p><p> </p><p>Also I'd suggest creating a base model and overriding the methods that are not same in both, just to avoid duplicate code.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/PyTorchLightning/pytorch-lightning/pull/4818/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Reply:
<p>Yes I will try <code>update_bn</code> in <code>on_train_end</code>.</p><p> </p><p>SWA is a method to improve generalization when using SGD. There's more details at https://pytorch.org/blog/stochastic-weight-averaging-in-pytorch/ and also in the original paper at https://arxiv.org/abs/1803.05407</p>

Reply:
Regarding inheritance, the SWAResnet inherits from LitResnet and overrides the methods that need change.

Reply:
Although it's correct but I think it can be done in a better way using `load_from_checkpoint` and keeping Model init same for both. Can I make a commit with some changes here by the weekend, if that's okay with you??

Reply:
Yeah sure! Go for it!

Reply:
> the SWAResnet inherits from LitResnet and overrides the methods that need change.

it was all good.

made some changes. BTW thanks for the SWA link.